app/validators/employment_details/answers_validator.rb

```diff
@@ -13,21 +14,24 @@ def initialize(record)
 
     def validate
       return unless applicable?
-      return if complete?
 
-      errors.add :employment_status, :incomplete
-      errors.add :base, :incomplete_records
-    end
+      errors.add :employment_status, :incomplete if record.employment_status.blank?
+      errors.add :employment_status, :incomplete unless not_working_details_complete?
 
-    def applicable?
-      requires_means_assessment?
+      validate_employment
+      validate_partner_employment
+
+      errors.add :base, :incomplete_records if errors.present?
     end
 
-    def complete?
-      return false if record.employment_status.blank?
-      return false if include_partner_in_means_assessment? && record.partner_employment_status.blank?
+    def validate_partner_employment
+      return unless include_partner_in_means_assessment?
 
-      not_working_details_complete?
+      errors.add :partner_employment_status, :incomplete if record.partner_employment_status.blank?
+    end
+
+    def applicable?
+      requires_means_assessment?
     end
 
     def not_working_details_complete?
```

Comment:
@RoseSAK What do you think of changing method name
```suggestion
def working_details_complete?
  income.date_job_lost.blank?
end
```